src/pudl/transform/eiaaeo.py

```diff
@@ -0,0 +1,324 @@
+    a less general subtopic, and specific variable name. It may also contain a
+    fourth field for a specific dimension such as fuel type.
+    """
+    fields = series_name.str.split(",", expand=True).rename(
+        columns={
+            0: "variable_fields",
+            1: "region_series",
+            2: "case_series",
+            3: "report_year_series",
+        }
+    )
+    variable_fields = fields.variable_fields.str.split(" : ", expand=True).rename(
+        columns={0: "topic", 1: "subtopic", 2: "variable_name", 3: "dimension"}
+    )
+    return fields.join(variable_fields)
+
+
+def get_category_info(category_name: pd.Series) -> pd.Series:
+    """Break human-readable category name into machine-readable fields.
+
+    Fortunately the only field we're pulling out of the category so far is the
+    region, which is the last of two comma-separated fields.
+    """
+    fields = category_name.str.rsplit(",", n=1).str.get(-1).rename("region_category")
+    return fields
+
+
+def filter_enrich_sanitize(
+    raw_df: pd.DataFrame, relevant_series_names: tuple[str]
+) -> pd.DataFrame:
+    """Basic cleaning steps common to all AEO tables.
+
+    1. Filter the AEO rows based on the series name
+    2. Break the series name and category names into useful fields
+    3. Sanitize strings & turn data values into a numeric field
+    4. Make some defensive checks about data from multiple sources that
+       *should* agree.
+    """
+    filtered = raw_df.loc[raw_df.series_name.str.startswith(relevant_series_names)]
+
+    enriched = pd.concat(
+        [
+            filtered,
+            get_series_info(filtered.series_name),
+            get_category_info(filtered.category_name),
+        ],
+        axis="columns",
+    ).drop(
+        columns=[
+            "series_name",
+            "category_name",
+            "variable_fields",
+        ]
+    )
+
+    sanitized = enriched.assign(
+        **{
+            colname: __sanitize_string(enriched[colname])
+            for colname in enriched.columns
+            if colname not in {"projection_year", "value"}
+        }
+    ).pipe(
+        lambda df: df.assign(
+            report_year_series=df.report_year_series.str.replace("aeo", ""),
+            value=pd.to_numeric(df.value.str.replace("- -", "")),
+        )
+    )
+
+    assert (
+        sanitized.region_series == sanitized.region_category
+    ).all(), "Series and category must agree on region!"
+    assert (
+        sanitized.case_series == sanitized.model_case_eiaaeo
+    ).all(), "Series and taxonomy must agree on case!"
+
+    # TODO 2024-04-20: one day we'll include report year in the extract step -
+    # at that point we'll have to compare that with the report year extracted
+    # from the series name.
+    return sanitized.drop(columns=["region_series", "case_series"]).rename(
+        columns={"region_category": "region", "report_year_series": "report_year"}
```

Comment:
two questions:
* could you just name the series in `get_category_info` `region` if you are going to just rename this down here?
* is this the emm region? why not name it with its long/pudl name as soon as you get it out of the category col?

Reply:
two answers:
* I compare the region from the category name & the series name to make sure they match - so I wanted to keep that origin suffix
* I found it a little nicer to have One Big Rename that renames everything to the final PUDL name rather than Several Small Renames when I was writing the code 🤷 Is this blocking?

Reply:
grand grand this makes sense and this is definitely not blocking! i personally like early renames for easier searching but that is 1000% a personal preference.